File: ThyroidClassificationProject/backend/src/main/java/com/thyroid/system/service/PatientService.java
package com.thyroid.system.service;

import com.thyroid.system.dto.PatientDTO;
import com.thyroid.system.model.Patient;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PatientService {
    PatientDTO createPatient(PatientDTO patientDTO, String createdBy);
    PatientDTO getPatientById(String id);
    Page<PatientDTO> getAllPatients(Pageable pageable);
    PatientDTO updatePatient(String id, PatientDTO patientDTO);
    void deletePatient(String id);
    Page<PatientDTO> searchPatients(String name, Pageable pageable);
}

File: frontend/src/services/patient.service.js
import api from "./api";

const getAllPatients = (page = 0, size = 10) => {
    return api.get(`/patients?page=${page}&size=${size}`);
};

const getPatient = (id) => {
    return api.get(`/patients/${id}`);
};

const createPatient = (data) => {
    return api.post("/patients", data);
};

const updatePatient = (id, data) => {
    return api.put(`/patients/${id}`, data);
};

const deletePatient = (id) => {
    return api.delete(`/patients/${id}`);
};

const searchPatients = (name, page = 0, size = 10) => {
    return api.get(`/patients/search?name=${name}&page=${page}&size=${size}`);
};

const PatientService = {
    getAll: getAllPatients,
    getAllPatients,
    getPatient,
    createPatient,
    updatePatient,
    deletePatient,
    searchPatients
};

export default PatientService;
